tasks/ast_codegen/src/generators/derive_clone_in.rs

```diff
@@ -0,0 +1,134 @@
+use itertools::Itertools;
+use proc_macro2::TokenStream;
+use quote::{format_ident, quote};
+use syn::Ident;
+
+use crate::{
+    output,
+    schema::{EnumDef, GetIdent, StructDef, TypeDef},
+    GeneratorOutput, LateCtx,
+};
+
+use super::{define_generator, generated_header, Generator};
+
+define_generator! {
+    pub struct DeriveCloneIn;
+}
+
+impl Generator for DeriveCloneIn {
+    fn name(&self) -> &'static str {
+        stringify!(DeriveCloneIn)
+    }
+
+    fn generate(&mut self, ctx: &LateCtx) -> GeneratorOutput {
+        let impls: Vec<TokenStream> = ctx
+            .schema
+            .definitions
+            .iter()
+            .filter(|def| def.generates_derive("CloneIn"))
+            .map(|def| match &def {
+                TypeDef::Enum(it) => derive_enum(it),
+                TypeDef::Struct(it) => derive_struct(it),
+            })
+            .collect();
+
+        let header = generated_header!();
+
+        GeneratorOutput::Stream((
+            output(crate::AST_CRATE, "derive_clone_in.rs"),
+            quote! {
+                #header
+
+                use oxc_allocator::{Allocator, CloneIn};
+                use crate::ast::*;
```

Comment:
I suggest adding an `endl!();` to stop rustfmt re-ordering these 2.

```suggestion
                use oxc_allocator::{Allocator, CloneIn};
                endl!();
                use crate::ast::*;
```